src/app/view/device/add-devices/add-devices.component.ts

```diff
@@ -405,33 +573,59 @@ export class AddDevicesComponent {
           }
         },
         error: (err) => {
-          //Error callback
+          // Error callback
           console.error('error caught in component', err.error.message);
           if (err.error.statusCode === 403) {
             this.toastrService.error(
               "You don't have the permissions to add a device.",
               'Access Denied',
             );
           } else {
+            this.submitButtonText = 'Submit';
+            this.isSubmitting = false;
             this.toastrService.error(
-              'some error occurred due to ' + err.error.message,
+              'Some error occurred due to ' + err.error.message,
               'Device!' + element.externalId,
             );
           }
         },
       });
     });
   }
+  shortenFileName(fileName: string, maxLength: number = 20): string {
+    if (!fileName || fileName.length <= maxLength) {
+      return fileName;
+    }
+
+    const extension = fileName.includes('.')
+      ? fileName.slice(fileName.lastIndexOf('.'))
+      : '';
+    const nameWithoutExtension = fileName.slice(0, fileName.lastIndexOf('.'));
+
+    const halfLength = Math.floor((maxLength - 3 - extension.length) / 2);
+    const start = nameWithoutExtension.slice(0, halfLength);
+    const end = nameWithoutExtension.slice(
+      nameWithoutExtension.length - halfLength,
+    );
+
+    return `${start}...${end}${extension}`;
+  }
   openPopupDialog() {
     this.dialogRef = this.dialog.open(this.popupDialog, {
       width: '700px',
     });
     this.dialogRef.afterClosed().subscribe((result: boolean) => {
       if (result) {
         this.submitForm();
+      } else {
+        this.isSubmitting = false;
       }
     });
   }
+  onAgreeClick() {
```

Comment:
Can't we name this function like `onSubmit`

Reply:
The file already has a function called onsubmit